qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationSymmetriseTab.cpp

```diff
@@ -156,31 +121,31 @@ void InelasticDataManipulationSymmetriseTab::setFileExtensionsByName(bool filter
   m_view->setWSSuffixes(filter ? getSampleWSSuffixes(tabName) : noSuffixes);
 }
 
-void InelasticDataManipulationSymmetriseTab::handleEnumValueChanged(QtProperty *prop, int value) {
-  if (prop->propertyName() == "ReflectType") {
-    m_model->setIsPositiveReflect(value == 0);
-  }
+void InelasticDataManipulationSymmetriseTab::handleReflectTypeChanged(int value) {
+  m_model->setIsPositiveReflect(value == 0);
 }
 
-void InelasticDataManipulationSymmetriseTab::handleDoubleValueChanged(QtProperty *prop, double value) {
-  if (prop->propertyName() == "Spectrum No") {
+void InelasticDataManipulationSymmetriseTab::handleDoubleValueChanged(std::string const &propName, double value) {
+  if (propName == "Spectrum No") {
     m_view->replotNewSpectrum(value);
   } else {
-    m_view->updateRangeSelectors(prop, value);
-    if (prop->propertyName() == "Elow") {
+    m_view->updateRangeSelectors(propName, value);
+    if (propName == "Elow") {
       m_model->getIsPositiveReflect() ? m_model->setEMin(value) : m_model->setEMax((-1) * value);
-    } else if (prop->propertyName() == "Ehigh") {
+    } else if (propName == "Ehigh") {
       m_model->getIsPositiveReflect() ? m_model->setEMax(value) : m_model->setEMin((-1) * value);
     }
   }
 }
 
-void InelasticDataManipulationSymmetriseTab::handleDataReady(QString const &dataName) {
+void InelasticDataManipulationSymmetriseTab::handleDataReady(std::string const &dataName) {
   if (m_view->validate()) {
     m_view->plotNewData(dataName);
   }
-  m_model->setWorkspaceName(dataName);
+  m_model->setWorkspaceName(QString(dataName.data()));
```

Comment:
Ideally we we want to keep all Qt including QString in the view. Would it be straightforward to refactor the model a little so that it uses std::string instead of QString? I also find it harder to read QString's when I'm debugging an issue! (Sorry this has fallen on you as I know this has been an existing issue!)

Reply:
No problem,  I agree with you.